src/objects/zcl_abapgit_object_aifc.clas.abap

```diff
@@ -0,0 +1,509 @@
+                                      ix_previous = lx_dyn_call_error ).
+      CATCH cx_root INTO lx_root.
+        RAISE EXCEPTION TYPE zcx_abapgit_exception
+          EXPORTING
+            previous = lx_root.
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD validate_interface.
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lx_root TYPE REF TO cx_root.
+
+    rv_success = abap_false.
+    TRY.
+        CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~VALIDATE_INTERFACE')
+          EXPORTING
+            is_ifkeys  = is_ifkeys
+          RECEIVING
+            rv_success = rv_success.
+
+      CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                      ix_previous = lx_dyn_call_error ).
+      CATCH cx_root INTO lx_root.
+        RETURN.
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+    DATA  lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+
+    DATA ls_icd_data_key TYPE ty_icd_data_key.
+    ls_icd_data_key-depl_scenario = ms_icd_data_key-depl_scenario.
+    ls_icd_data_key-ns = ms_icd_data_key-ns.
+    ls_icd_data_key-ifname = ms_icd_data_key-ifname.
+    ls_icd_data_key-ifver2 = ms_icd_data_key-ifver2.
+
+    CLEAR rv_user.
+
+    TRY.
+        CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~CHANGED_BY')
+          EXPORTING
+            is_key  = ls_icd_data_key
+          RECEIVING
+            rv_user = rv_user.
+
+      CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                      ix_previous = lx_dyn_call_error ).
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+    zcx_abapgit_exception=>raise( 'Delete not supported.'(001) ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+    DATA: lx_root TYPE REF TO cx_root.
+    DATA: lt_content TYPE ty_content_t.
+
+    DATA lr_tabledescr TYPE REF TO cl_abap_tabledescr.
+    DATA lr_structdescr TYPE REF TO cl_abap_structdescr.
+    DATA lr_table TYPE REF TO data.
+    FIELD-SYMBOLS <lt_table> TYPE STANDARD TABLE.
+    FIELD-SYMBOLS <ls_table> TYPE any.
+
+    DATA ls_ifkey TYPE ty_aif_key_s.
+    DATA lr_content TYPE REF TO ty_content_s.
+
+    DATA lv_tablename TYPE string.
+    FIELD-SYMBOLS: <lv_value> TYPE any.
+
+    IF iv_step <> zif_abapgit_object=>gc_step_id-abap.
+      RETURN.
+    ENDIF.
+
+    TRY.
+        io_xml->read( EXPORTING
+                        iv_name = `Content_table`
+                      CHANGING
+                        cg_data = lt_content ).
+
+
+        LOOP AT lt_content REFERENCE INTO lr_content.
+          TRY.
+              lv_tablename = cl_abap_dyn_prg=>check_table_name_str( val = lr_content->tabname
+                                                                    packages = '' ).
+            CATCH cx_abap_not_a_table.
+              RAISE EXCEPTION TYPE zcx_abapgit_exception.
```

Comment:
suggest to use `zcx_abapgit_exception=>raise` instead of `RAISE EXCEPTION TYPE zcx_abapgit_exception.` or supply ix_previous. Because the latter will only show a generic 'An exception was raised' to the user which is usually not very helpful.